### PR TITLE
don't syntax highlight hidden pre tags

### LINKED ISSUE #2474: Don't syntax highlight hidden <pre> tags
Look at https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Colors/Color_picker_tool for example. 
[Its raw HTML](https://github.com/mdn/content/blob/main/files/en-us/web/css/css_colors/color_picker_tool/index.html) looks something like this:

```html
<div style="display: none;">
<h2 id="ColorPicker_Tool">ColorPicker tool</h2>

<h3 id="HTML">HTML</h3>

<pre class="brush: html">    &lt;div id="container"&gt;
        &lt;div id="palette" class="block"&gt;
            &lt;div id="color-palette"&gt;&lt;/div&gt;
...

```

But when Yari gets its hands on it, that `<pre>` tag is extracted and sent into Node Prism to be syntax highlighted. 
What happens is that the HTML bloats unnecessarily for `<pre>` tags that are never meant to be consumed by the human eye. See view-source:https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Colors/Color_picker_tool whose HTML is serialized for in the HTML itself and then in the JSON serialized React state (towards the end). This slows down Prism syntax highlighting unnecessarily and most importantly, it makes the HTML bigger than it needs to be.

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #2474

This is a DRAFT because I'm not sure it's "safe" to proceed. ("safe" as a wise, not safe as in security)

Basically, what this PR does is that it does *not* bother doing Prism syntax highlighting of `<pre class="brush: xxx">` tags that are located inside a `<div class="hidden">`. These `pre` tags aren't meant for human eye consumption anyway. They exist to be the source of the embedded live samples. Being able to skip the syntax highlighting has the following advantages:

1. Less build-time work to do in the Yari builder (least important)
2. The HTML is smaller
3. Because the HTML itself is serialized as JSON, for the React `<Document>` component, the serialized JSON also becomes smaller (aka. the "double whammy")
4. The `index.json`, which contains the rendered HTML blobs becomes smaller
5. Not just in terms of network bytes to transfer, also less CPU work necessary by the client

For example, I looked at http://localhost:5000/en-US/docs/Web/CSS/CSS_Colors/Color_picker_tool/ and http://localhost:3000/en-US/docs/Learn/CSS/Styling_text/Fundamentals and did some experimentation to look at how the size of the HTML and JSON changes with these changes:
https://docs.google.com/spreadsheets/d/1jftCONH3Bs5HyNKe1YSnDJBX3tFbjJpQ6S4Sy_l8uRU/edit?usp=sharing

I can't do a complete build because there are still lots of `<div style="display:none">` in the raw sources, that needs to be changed to `<div class="hidden">` but the two above-mentioned URLs used `<div class="hidden">`.
Focusing on http://localhost:3000/en-US/docs/Learn/CSS/Styling_text/Fundamentals first, the conclusion of the spreadsheet is that:

1. The basic solution is a 56KB improvement in HTML and 1KB improvement in gzipped HTML
2. The extreme solution is a 72KB improvement in HTML and 2.7KB improvement in gzipped HTML

The improvement for that ColorPicker page is much much more.
The improvement for the JSON files was a bit less but it's also less important.

What's the difference between the "basic solution" and the "extreme solution". 
The **basic solution** just omits syntax highlighting the hidden `pre` tags but leaves them in the HTML. 
The **extreme solution** not only omits the syntax highlighting, it also entirely deletes the `<pre>` tag because once the live samples have been generated (based on the kumascript rendered HTML), it's no longer needed. 
For a really curious reader, to find out the source for the live samples, they'd have to click the "Edit source in GitHub" to go back to the original `index.html` source to see, rather than being able to use `view-source:/`.

### CODE REVIEW

**Comment by peterbe** [comment]
[This one](https://github.com/mdn/content/pull/1269) will make the results even better.

**Comment by peterbe** [comment]
I ran a complete build (all 11.5k pages) before and after with this branch. 
The size difference isn't particularly impressive. 
Before: 1.3GB
After: 1.2GB

But that's less important. If that change makes reading the page faster, it makes reading the page better, it makes the user happier.

**Comment by peterbe** [comment]
@escattone Can you kill two birds with one stone? Can you figure out if it's perfect same and safe to essentially remove the hidden `<pre>` tags once the live samples have gotten the HTML it needs to create the `_samples_/index.html` file?
Also, to review the code itself. 
Once we know about that, I'll tidy up the PR when we know which strategy is best.

**Comment by peterbe** [comment]
@escattone Another good reason why we should strip out any `div.hidden pre` (after the live samples have been extracted) is because otherwise the flaw detection will scrutinize those `pre` tags. Right now, I don't think there's anything terrible going on in there but the hidden `<pre>` tags used for live samples might be very different from the ones meant to be displayed. So if we do things like Prettier checking, it might complain about `pre` tags that are never meant to be shown.

**Comment by peterbe** [comment]
Actually, I've gone ahead and done it now. Instead of skipping the hidden `pre` tags, it just removes them entirely before proceeding into the syntax highlighting. 

Fun fact, the HTML of http://localhost:3000/en-us/docs/learn/css/styling_text/fundamentals becomes 77KB smaller compared to https://developer.mozilla.org/en-us/docs/learn/css/styling_text/fundamentals as of this PR.

**Review by escattone — APPROVED** [review]
@peterbe This is a great idea. It's simple, it works without breaking anything, and it provides some tangible benefits. 👍

File: build/syntax-highlight.js [hunk 1]
```diff
@@ -50,10 +50,16 @@ const IGNORE = new Set(["none", "text", "plain", "unix"]);
 function syntaxHighlight($, doc) {
   loadAllLanguages();
 
+  // Our content will be like this: `<pre class="brush:js">` or
+  // `<pre class="brush: js">` so we're technically not looking for an exact
+  // match. The wildcard would technically match `<pre class="brushetta">`
```
[the remark this answers is not in the document]
Comment (mattiapontonio) [reply build/syntax-highlight.js:55]: It's _bruschetta_ 🙂🍕

[the remark this answers is not in the document]
Comment (peterbe) [reply build/syntax-highlight.js:55]: But had I got it right it would have missed the point :)

**Pull request merged**